modules/nf-core/gprofiler2/gost/main.nf

```diff
@@ -0,0 +1,30 @@
+process GPROFILER2_GOST {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "conda-forge::r-ggplot2=3.4.3 conda-forge::r-gprofiler2=0.2.2"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-3712554873398d849d0d11b22440f41febbc4ede:aa19bb8afc0ec6456a4f3cd650f7577c3bbdd4f3-0':
+        'biocontainers/mulled-v2-3712554873398d849d0d11b22440f41febbc4ede:aa19bb8afc0ec6456a4f3cd650f7577c3bbdd4f3-0' }"
+
+    input:
+    tuple val(meta), path(de_file)
+    path(background_file)
+
+    output:
+    tuple val(meta), path("*gprofiler2.*all_enriched_pathways.tsv") , emit: all_enrich
+    tuple val(meta), path("*gprofiler2.*gost_results.rds")          , emit: rds         , optional: true
+    tuple val(meta), path("*gprofiler2.*gostplot.png")              , emit: plot_png    , optional: true
+    tuple val(meta), path("*gprofiler2.*gostplot.html")             , emit: plot_html   , optional: true
+    tuple val(meta), path("*gprofiler2.*sub_enriched_pathways.tsv") , emit: sub_enrich  , optional: true
+    tuple val(meta), path("*gprofiler2.*sub_enriched_pathways.png") , emit: sub_plot    , optional: true
+    tuple val(meta), path("*ENSG_filtered.gmt")                     , emit: filtered_gmt, optional: true
+    tuple val(meta), path("*R_sessionInfo.log")                     , emit: session_info
+    path "versions.yml"                                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    template 'gprofiler2_gost.R'
```

Comment:
I am also unsure about template calling in nf-core, so again, would need a second reviewer on this.

Reply:
This is done in some other modules as well, like here :)
https://github.com/nf-core/modules/blob/master/modules/nf-core/deseq2/differential/main.nf